### DISCUSSION TITLE
LPUART Interrupt waiting, program hang on

### DISCUSSION THREAD

**Discussion by YMont** [zephyrproject-rtos/zephyr#70116] [Q&A] [answered]
**Question**
Hi,
My question is that program hang on waiting **RIE**, **ILIE**, **ORIE** mask, why they are waiting here?
I read specification RT1060 ,but I don't understand what is mean? Please explain or give some suggestion for it.
Thanks a lot.

**Code**
`modules\hal\nxp\mcux\mcux-sdk\drivers\lpuart\fsl_lpuart.c`
```c
/* Disable and re-enable the global interrupt to protect the interrupt enable register during
 * read-modify-wrte. */
irqMask = DisableGlobalIRQ();
/* Enable RX interrupt. */
base->CTRL |= (uint32_t)(LPUART_CTRL_RIE_MASK | LPUART_CTRL_ILIE_MASK | LPUART_CTRL_ORIE_MASK);
 EnableGlobalIRQ(irqMask);
```

**Comment by nashif** [comment]
@dleach02 @danieldegrasse FYI

**Comment by DerekSnell** [comment, marked as answer]
Hi @YMont ,
Those three bits are interrupt enables for the LPUART peripheral.  Those bits, and all registers are documented in the [RT1060 Reference manual](https://www.nxp.com/webapp/Download?colCode=IMXRT1060RM), see the CTRL register description in the LPUART chapter.

And the code you referenced is not waiting those bits.  That line sets the 3 enables, and then enables interrupts.

Perhaps you could provide more details about the issue you are facing.  Are you using one of the Zephyr samples in the repo on one of the [RT1060 EVK boards](https://docs.zephyrproject.org/latest/boards/nxp/mimxrt1060_evk/doc/index.html)?

Best regards
